sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Azure.Storage.Files.Shares.Perf.csproj

```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>1.0.0</Version>
+    <OutputType>Exe</OutputType>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Storage.Files.Shares" />
```

Comment:
Here's why I feel it is better to use `PackageReference` for perf tests of a public API.
1. We can lock the version and ensure that we are testing a particular snapshot of the service. We can rule out a change in the service when looking at any change in numbers.
1. We can easily compare perf of different versions of the service at any time by changing the ref.
1. Referencing the source directly makes the thing we are trying to measure a moving target. Plus, it is difficult to rationalize a change in numbers. We have to spend time investigating and making sure the source was not changed in any way.
1. Functional tests are more of a live test of the source code.

There are some advantages to using project references. Let me know what you think.

Reply:
The biggest problem with the PackageReference is that it uses the centralized version that is very rarely updated (by design) so the perf test would always be measuring some old version of the package and not the code that's going to ship.

For example the current version of Azure.Storage.Files.Shares package reference in 12.0.1. It's almost a year old. https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Packages.Data.props#L29

While the latest released is 12.5.

Being able to have a baseline and compare packages and run against different versions is very important. I think we would need to, eventually, build more infrastructure around it.

What I want to avoid is releasing a package with a huge perf regression while having a perf test because we were never benchmarking the shipping code.